backend/infrastructure/repository/article_repositoy.go

```diff
@@ -50,10 +50,97 @@ func (repo *GormArticleRepository) GetArticlesAll() ([]article.Article, error) {
 	return repo.getArticlesByCondition("1 = 1")
 }
 
+func (repo *GormArticleRepository) GetArticles(limit, offset int) ([]article.Article, error) {
+	return repo.getArticlesWithLimitOffset("1 = 1", limit, offset)
+}
+
 func (repo *GormArticleRepository) GetArticlesByCategoryID(categoryID int) ([]article.Article, error) {
 	return repo.getArticlesByCondition("category_id = ?", categoryID)
 }
 
+// カテゴリ名とタグ名を指定して記事を取得
+func (repo *GormArticleRepository) GetArticlesByCategoryAndTag(categoryName, tagName string, limit, offset int) ([]article.Article, error) {
+	if categoryName != "" && tagName != "" {
+		return repo.GetArticlesByCategoryTagName(categoryName, tagName, limit, offset)
+	}
+	if categoryName != "" {
+		return repo.GetArticlesByCategoryName(categoryName, limit, offset)
+	}
+	if tagName != "" {
+		return repo.GetArticlesByTagName(tagName, limit, offset)
+	}
+	return repo.GetArticles(limit, offset)
+}
+
+func (repo *GormArticleRepository) GetArticlesByCategoryName(categoryName string, limit, offset int, args ...interface{}) ([]article.Article, error) {
+	var articles []Article
+	err := repo.db.DB.
+		Joins("JOIN categories ON articles.category_id = categories.id").
+		Where("categories.name = ?", categoryName). // カテゴリ名で絞り込み
+		Order("articles.created_at DESC").          // created_atの降順でソート
+		Limit(limit).
+		Offset(offset).
+		Preload("Category").
+		Preload("Tags").
+		Find(&articles).Error
+	if err != nil {
+		return nil, err
+	}
+
+	var domainArticles []article.Article
```

Comment:
他のところもそうですが、ここはメモリ効率を考えるとこうした方が良かったような記憶があります。capを超えると毎回アロケートされるような物があった記憶があります。
```suggestion
	domainArticles := make([]article.Article, 0, len(articles))
```
mustではないです！ぶっちゃけ、あまり問題にならないレベルだと思います